package/upgrade/lib.sh

```diff
@@ -571,6 +571,57 @@ EOF
   wait_for_addon_upgrade_deployment $name $namespace $enabled $curstatus
 }
 
+upgrade_harvester_upgradelog_with_patch_loggingref()
+{
+  local namespace=$UPGRADE_NAMESPACE
```

Comment:
Variables should be qouted with `"`, unless you want to risk expansion and globbing.

I can only recommend to install [shellcheck](https://github.com/koalaman/shellcheck) as a linter for shellscripts in your IDE. It will really help your write more robust shellscripts.

Reply:
Updated, thanks.